users/mkillewald/myfork_keychron_ft.c

```diff
@@ -0,0 +1,271 @@
+        case RESET_KEY2_ALT:
+#    endif
+            if (record->event.pressed) {
+                key_press_status |= KEY_PRESS_STEP_2;
+                if (key_press_status == KEY_PRESS_FACTORY_RESET) {
+                    timer_3s_buffer = sync_timer_read32() == 0 ? 1 : sync_timer_read32();
+                }
+            } else {
+                key_press_status &= ~KEY_PRESS_STEP_2;
+                timer_3s_buffer = 0;
+            }
+            return true;
+#endif
+#if defined(BL_TEST_KEY1) && defined(BL_TEST_KEY2)
+        case BL_TEST_KEY1:
+            if (record->event.pressed) {
+                key_press_status |= KEY_PRESS_STEP_3;
+                if (led_test_mode) {
+                    if (++led_test_mode >= LED_TEST_MODE_MAX) {
+                        led_test_mode = LED_TEST_MODE_WHITE;
+                    }
+                } else if (key_press_status == KEY_PRESS_LED_TEST) {
+                    timer_3s_buffer = sync_timer_read32() == 0 ? 1 : sync_timer_read32();
+                }
+            } else {
+                key_press_status &= ~KEY_PRESS_STEP_3;
+                timer_3s_buffer = 0;
+            }
+            return true;
+        case BL_TEST_KEY2:
+#    if defined(BL_TEST_KEY2_ALT1)
+        case BL_TEST_KEY2_ALT1:
+#    endif
+#    if defined(BL_TEST_KEY2_ALT2)
+        case BL_TEST_KEY2_ALT2:
+#    endif
+            if (record->event.pressed) {
+                key_press_status |= KEY_PRESS_STEP_4;
+                if (led_test_mode) {
+                    led_test_mode = LED_TEST_MODE_OFF;
+#    if defined(RGB_MATRIX_ENABLE)
+                    if (is_win_mode()) {
+                        set_win_base_rgb();
+                    } else {
+                        set_mac_base_rgb();
+                    }
+#    endif
+                } else if (key_press_status == KEY_PRESS_LED_TEST) {
+                    timer_3s_buffer = sync_timer_read32() == 0 ? 1 : sync_timer_read32();
+                }
+            } else {
+                key_press_status &= ~KEY_PRESS_STEP_4;
+                timer_3s_buffer = 0;
+            }
+            return true;
+#endif
+        default:
+            return true;
+    }
+}
+
+static void factory_reset(void) {
+    timer_300ms_buffer = sync_timer_read32() == 0 ? 1 : sync_timer_read32();
+    factory_reset_count++;
+    layer_state_t default_layer = default_layer_state;
+    eeconfig_init();
+    default_layer_set(default_layer);
+    led_test_mode = LED_TEST_MODE_OFF;
+#ifdef LED_MATRIX_ENABLE
+    if (!led_matrix_is_enabled()) {
+        led_matrix_enable();
+    }
+    led_matrix_init();
+#endif
+#ifdef RGB_MATRIX_ENABLE
+    if (!rgb_matrix_is_enabled()) {
+        rgb_matrix_enable();
+    }
+    rgb_matrix_init();
+#    ifdef SPLIT_KEYBOARD
+    led_state = rgb_matrix_get_mode();
+    hsv       = rgb_matrix_get_hsv();
+    rgb_matrix_mode_noeeprom(RGB_MATRIX_SOLID_COLOR);
+#    endif
+#endif
+}
+
+static void timer_3s_task(void) {
```

Comment:
you may want to look at the deferred executor feature.  It does a lot of what you're trying to do here. 

https://docs.qmk.fm/#/custom_quantum_functions?id=deferred-execution

Reply:
Was not aware this existed, so thanks for pointing it out!